lib/widgets/witnet/transactions/value_transfer/create_dialog_box/vtt_builder/01_recipient_step.dart

```diff
@@ -180,6 +187,52 @@ class RecipientStepState extends State<RecipientStep>
     );
   }
 
+  _activateQrScanner(BuildContext context, ThemeData theme) {
+    Navigator.push(
+      context,
+      MaterialPageRoute(
+        builder: (context) => QrScanner(
+          currentRoute: CreateVttScreen.route,
+          onChanged: (String value) => {
+            Navigator.popUntil(
+                context, ModalRoute.withName(CreateVttScreen.route)),
+            _addressController.text = value,
+            setAddress(value),
+          },
+        ),
+      ),
+    );
+  }
+
+  _recipientAddressTextFormField(BuildContext context, ThemeData theme) {
+    return TextFormField(
+      style: theme.textTheme.bodyLarge,
+      decoration: InputDecoration(
+        hintText: 'Recipient address',
+        suffixIcon: QrScanner.iconButton(
```

Comment:
I think this should be part of the input not part of the qr scanner, if you want to separate the logic you can do it in a separate widget